packages/sdk/src/ParadymWalletSdk.tsx

```diff
@@ -0,0 +1,79 @@
+import { type FullAgent, type SetupAgentOptions, setupAgent } from './agent'
+import { ParadymWalletMustBeInitializedError } from './error'
+import { useCredentials } from './hooks/useCredentials'
+import { useCredentialsById } from './hooks/useCredentialsById'
+import { AgentProvider, useAgent } from './providers/AgentProvider'
+
+/**
+ *
+ * Options that will be used when initializing the Paradym Wallet SDK
+ *
+ * Make sure to instantiate the `ParadymWalletSdk` once and use the same instance within the application, via for example a `useContext`
+ *
+ */
+export type ParadymWalletSdkOptions = {
+  /**
+   *
+   * Options that setup the internal agent.
+   *
+   * You can define here whether you want to use `openid4vc`, `didcomm` or both
+   *
+   * @see `SetupAgentOptions` for the specific details
+   *
+   */
+  walletOptions: SetupAgentOptions
+}
+
+export class ParadymWalletSdk {
+  private agent: FullAgent
+
+  public constructor(options: ParadymWalletSdkOptions) {
+    this.agent = setupAgent(options.walletOptions) as FullAgent
+  }
+
+  private assertAgentIsInitialized() {
+    if (!this.agent.isInitialized) throw new ParadymWalletMustBeInitializedError()
+  }
+
+  /**
+   *
+   * Initialized the wallet sdk and sets everything up for usage
+   *
+   * @note this must be called before any usage of the agent, such as retrieving credentials
+   *
+   */
+  public async initialize() {
+    await this.agent.initialize()
+  }
+
+  /**
+   *
+   * All available hooks provided by the wallet SDK
+   *
+   * @todo Ideally, we only want to return the hooks based on whether it is a didcomm/openid4vc agent
+   *       This can be done quite simple, but in order to provide correct types, it get very complex
+   *
+   */
+  public get hooks() {
+    this.assertAgentIsInitialized()
+
+    return {
+      useAgent,
```

Comment:
Maybe we can split these between `hooks` and `internalHooks`. And we make `internalHooks` not dependant on semver / are more advanced. 

We recommend users to use the `hooks`, and fallback to `internalHooks` with caution.

`useAgent` will be in `internalHooks`.